src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1942,6 +1948,36 @@ public void AddFromAliasToWhereTokens(string fromAlias)
             }
         }
 
+        public void AddFromAliasToOrderByTokens(string fromAlias)
+        {
+            if (string.IsNullOrEmpty(fromAlias))
+                throw new InvalidOperationException("Alias cannot be null or empty");
+
+            var tokens = GetCurrentOrderByTokens();
+            var current = tokens.First;
+            while (current != null)
+            {
+                if (current.Value is OrderByToken w)
+                    current.Value = w.AddAlias(fromAlias);
+                current = current.Next;
+            }
+        }
+
+        public void AddFromAliasToFilterTokens(string fromAlias)
+        {
+            if (string.IsNullOrEmpty(fromAlias))
+                throw new InvalidOperationException("Alias cannot be null or empty");
+
+            var tokens = GetCurrentFilterTokens();
+            var current = tokens.First;
+            while (current != null)
+            {
+                if (current.Value is WhereToken w)
+                    current.Value = w.AddAlias(fromAlias);
+                current = current.Next;
+            }
```

Comment:
You have the same code in all 3 methods - `AddFromAliasToXXXTokens`. Please try to avoid code duplication.

Reply:
Fixed